lib/galaxy/jobs/runners/util/job_script/__init__.py

```diff
@@ -124,18 +126,21 @@ def _handle_script_integrity(path, config):
                 script_integrity_verified = True
                 break
 
+            log.debug("Script integrity error: returncode was %d", returncode)
+
             # Else we will sync and wait to see if the script becomes
             # executable.
             try:
                 # sync file system to avoid "Text file busy" problems.
                 # These have occurred both in Docker containers and on EC2 clusters
                 # under high load.
                 subprocess.check_call(INTEGRITY_SYNC_COMMAND)
-            except Exception:
-                pass
+            except Exception as e:
+                log.debug("Error syncing the filesystem: %s", unicodify(e))
+
             time.sleep(sleep_amt)
-        except Exception:
-            pass
+        except Exception as exc:
+            log.debug("Script not available yet: %s", unicodify(exc))
```

Comment:
Can you add a sleep here ? This will just cycle very rapidly through the for loop.
```
galaxy.jobs.runners.util.job_script: DEBUG: Script not available yet: [Errno 26] Text file busy
galaxy.jobs.runners.util.job_script: DEBUG: Script not available yet: [Errno 26] Text file busy
galaxy.jobs.runners.util.job_script: DEBUG: Script not available yet: [Errno 26] Text file busy
galaxy.jobs.runners.util.job_script: DEBUG: Script not available yet: [Errno 26] Text file busy
galaxy.jobs.runners: DEBUG: (170) command is: rm -rf working; mkdir -p working; cd working; /tmp/tmpp_sF2R/tmpw89Dhd/tmphkVELS/database/job_working_directory_XR19i2/000/170/tool_script.sh; return_code=$?; cd '/tmp/tmpp_sF2R/tmpw89Dhd/tmphkVELS/database/job_working_directory_XR19i2/000/170'; 
[ "$GALAXY_VIRTUAL_ENV" = "None" ] && GALAXY_VIRTUAL_ENV="$_GALAXY_VIRTUAL_ENV"; _galaxy_setup_environment True
python "/tmp/tmpp_sF2R/tmpw89Dhd/tmphkVELS/database/job_working_directory_XR19i2/000/170/set_metadata_kPziaV.py" "/tmp/tmpp_sF2R/tmpw89Dhd/tmphkVELS/database/job_working_directory_XR19i2/000/170/registry.xml" "/tmp/tmpp_sF2R/tmpw89Dhd/tmphkVELS/database/job_working_directory_XR19i2/000/170/working/galaxy.json" "/tmp/tmpp_sF2R/tmpw89Dhd/tmphkVELS/database/job_working_directory_XR19i2/000/170/metadata_in_HistoryDatasetAssociation_178_wJ5pP7,/tmp/tmpp_sF2R/tmpw89Dhd/tmphkVELS/database/job_working_directory_XR19i2/000/170/metadata_kwds_HistoryDatasetAssociation_178_61XxNB,/tmp/tmpp_sF2R/tmpw89Dhd/tmphkVELS/database/job_working_directory_XR19i2/000/170/metadata_out_HistoryDatasetAssociation_178_xNFL_j,/tmp/tmpp_sF2R/tmpw89Dhd/tmphkVELS/database/job_working_directory_XR19i2/000/170/metadata_results_HistoryDatasetAssociation_178_6ZpSVA,/tmp/tmpp_sF2R/tmpw89Dhd/tmphkVELS/database/files/000/dataset_178.dat,/tmp/tmpp_sF2R/tmpw89Dhd/tmphkVELS/database/job_working_directory_XR19i2/000/170/metadata_override_HistoryDatasetAssociation_178_fPqz65" 5242880; sh -c "exit $return_code"
galaxy.jobs.runners.util.job_script: DEBUG: Script not available yet: [Errno 26] Text file busy
galaxy.jobs.runners.local: DEBUG: (170) executing job script: /tmp/tmpp_sF2R/tmpw89Dhd/tmphkVELS/database/job_working_directory_XR19i2/000/170/galaxy_170.sh
galaxy.jobs.runners.util.job_script: DEBUG: Script not available yet: [Errno 26] Text file busy
galaxy.jobs: DEBUG: (170) Persisting job destination (destination id: local:///)
galaxy.jobs.runners.util.job_script: DEBUG: Script not available yet: [Errno 26] Text file busy
galaxy.jobs.runners.util.job_script: DEBUG: Script not available yet: [Errno 26] Text file busy
galaxy.jobs.runners.util.job_script: DEBUG: Script not available yet: [Errno 26] Text file busy
galaxy.jobs.runners.util.job_script: DEBUG: Script not available yet: [Errno 26] Text file busy
galaxy.jobs.runners.util.job_script: DEBUG: Script not available yet: [Errno 26] Text file busy
galaxy.jobs.runners.util.job_script: DEBUG: Script not available yet: [Errno 26] Text file busy
galaxy.jobs.runners.util.job_script: DEBUG: Script not available yet: [Errno 26] Text file busy
galaxy.jobs.runners.util.job_script: DEBUG: Script not available yet: [Errno 26] Text file busy
galaxy.jobs.runners.util.job_script: DEBUG: Script not available yet: [Errno 26] Text file busy
galaxy.jobs.runners.util.job_script: DEBUG: Script not available yet: [Errno 26] Text file busy
galaxy.jobs.runners.util.job_script: DEBUG: Script not available yet: [Errno 26] Text file busy
galaxy.jobs.runners.util.job_script: DEBUG: Script not available yet: [Errno 26] Text file busy
galaxy.jobs.runners.util.job_script: DEBUG: Script not available yet: [Errno 26] Text file busy
galaxy.jobs.runners.util.job_script: DEBUG: Script not available yet: [Errno 26] Text file busy
galaxy.jobs.runners.util.job_script: DEBUG: Script not available yet: [Errno 26] Text file busy
requests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost
galaxy.jobs.runners.local: DEBUG: execution finished: /tmp/tmpp_sF2R/tmpw89Dhd/tmphkVELS/database/job_working_directory_XR19i2/000/167/galaxy_167.sh
galaxy.jobs.runners.util.job_script: DEBUG: Script not available yet: [Errno 26] Text file busy
galaxy.jobs.runners.util.job_script: DEBUG: Script not available yet: [Errno 26] Text file busy
galaxy.jobs.runners.util.job_script: DEBUG: Script not available yet: [Errno 26] Text file busy
galaxy.jobs.runners.util.job_script: DEBUG: Script not available yet: [Errno 26] Text file busy
galaxy.jobs.runners.util.job_script: DEBUG: Script not available yet: [Errno 26] Text file busy
galaxy.jobs.runners.util.job_script: DEBUG: Script not available yet: [Errno 26] Text file busy
galaxy.jobs.runners.util.job_script: DEBUG: Script not available yet: [Errno 26] Text file busy
galaxy.jobs.runners.util.job_script: DEBUG: Script not available yet: [Errno 26] Text file busy
galaxy.jobs.runners.util.job_script: DEBUG: Script not available yet: [Errno 26] Text file busy
requests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/76bc6b8e6396dad0?key=c2c1276111cfc9584c170781285a636e HTTP/1.1" 200 None
galaxy.jobs.runners.util.job_script: DEBUG: Script not available yet: [Errno 26] Text file busy
galaxy.jobs.runners.util.job_script: DEBUG: Script not available yet: [Errno 26] Text file busy
galaxy.jobs.runners.util.job_script: DEBUG: Script not available yet: [Errno 26] Text file busy
galaxy.model.metadata: DEBUG: loading metadata from file for: HistoryDatasetAssociation 175
galaxy.jobs.runners.util.job_script: DEBUG: Script not available yet: [Errno 26] Text file busy
requests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost
galaxy.jobs.runners.util.job_script: DEBUG: Script not available yet: [Errno 26] Text file busy
galaxy.jobs.runners: ERROR: (169) Unhandled exception calling queue_job
Traceback (most recent call last):
  File "/galaxy/lib/galaxy/jobs/runners/__init__.py", line 104, in run_next
    method(arg)
  File "/galaxy/lib/galaxy/jobs/runners/local.py", line 87, in queue_job
    command_line, exit_code_path = self.__command_line(job_wrapper)
  File "/galaxy/lib/galaxy/jobs/runners/local.py", line 76, in __command_line
    self.write_executable_script(job_file, job_file_contents)
  File "/galaxy/lib/galaxy/jobs/runners/__init__.py", line 327, in write_executable_script
    write_script(path, contents, self.app.config, mode=mode)
  File "/galaxy/lib/galaxy/jobs/runners/util/job_script/__init__.py", line 112, in write_script
    _handle_script_integrity(path, config)
  File "/galaxy/lib/galaxy/jobs/runners/util/job_script/__init__.py", line 146, in _handle_script_integrity
    raise Exception("Failed to write job script, could not verify job script integrity.")
Exception: Failed to write job script, could not verify job script integrity.
```

Reply:
I was also going to suggest to move:
```
time.sleep(sleep_amt)
```
from line 141 to here. That would basically have the same effect of https://github.com/galaxyproject/galaxy/pull/4718.

Reply:
merge @mvdbeek's then, I'll rebase mine accordingly.

Reply:
I think it's better with the nested try/except as here, but I can push to your branch if you want.